src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,365 @@
+                }
+            }
+            if ("\n" === $this->data[$this->cursor]) {
+                throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+            }
+            ++$this->cursor;
+            $this->skipComment();
+            $value = str_replace(array('\\\\', '\\"', '\r', '\n'), array('\\', '"', "\r", "\n"), $value);
+        } else {
+            $notQuoted = true;
+            while ($this->cursor < $this->end && "\n" !== $this->data[$this->cursor] && !(' ' === $this->data[$this->cursor - 1] && '#' === $this->data[$this->cursor])) {
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+            }
+            $value = rtrim($value);
+            $this->skipComment();
+        }
+
+        $this->skipEmptyLines();
+
+        $currentValue = $value;
+        if (!$singleQuoted) {
+            $value = $this->resolveVariables($value);
+            $value = $this->resolveCommands($value);
+        }
+
+        if ($notQuoted && $currentValue == $value && preg_match('/\s+/', $value)) {
+            throw new FormatException('A value containing spaces must be surrounded by quotes', $this->createFormatExceptionContext());
+        }
+
+        return $value;
+    }
+
+    private function skipWhitespace()
+    {
+        $this->cursor += strspn($this->data, ' ', $this->cursor);
+    }
+
+    private function skipEmptyLines()
+    {
+        if (preg_match('/(\n+|^#[^\n]*(\n*|$))+/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function skipComment()
+    {
+        if (preg_match('/ *#[^\n]*(\n*|$)/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function resolveCommands($value)
+    {
+        if (false === strpos($value, '$')) {
+            return $value;
+        }
+
+        $regex = '/
+            (\\\\)?               # escaped with a backslash?
+            \$
+            (?<cmd>
+                \(                # require opening parenthesis
+                ([^()]|\g<cmd>)+  # allow any number of non-parens, or balanced parens (by nesting the <cmd> expression recursively)
+                \)                # require closing paren
+            )
+        /x';
+
+        $env = array_replace($_ENV, $this->values);
+        $path = $this->path;
+        $lineno = $this->lineno;
+
+        return preg_replace_callback($regex, function ($matches) use ($env, $path, $lineno) {
+            if ('\\' === $matches[1]) {
+                return substr($matches[0], 1);
+            }
+
+            if (!class_exists('Symfony\Component\Process\Process')) {
+                throw new \LogicException('Resolving commands requires the Symfony Process component.');
+            }
+
+            $process = new Process('echo '.$matches[0], null, $env);
```

Comment:
$_ENV, from which $env is sourced, is not populated by default on eg Ubuntu.
But Process 3.2 has a method to enable env inheriting, so that $env can be merged automatically.

why "echo" btw?